File: closestPair.swift
import Foundation

func distance(x1: Int, y1: Int, x2: Int, y2: Int) -> Double{
    return sqrt(Double((x2-x1)*(x2-x1) + (y2-y1)*(y2-y1)))
}

    if let n = Int(readLine()!){
        var points = [(Int, Int)]()

        for _ in 0..<n{
            let input = readLine()!.split(separator: " ").map{Int($0)!}
            points.append((input[0], input[1]))
        }

        var minDist = Double.greatestFiniteMagnitude

        for i in 0..<n{
            for j in i+1..<n{
                let dist = distance(x1: points[i].0, y1:points[i].1, x2:points[j].0, y2:points[j].1)
                minDist = min(minDist, dist)
            }
        }
        print(String(format: "%.14f", minDist))
    }

File: closestPair.c
#include <stdio.h>
#include <math.h>
#include <float.h>

double distance(int x1, int y1, int x2, int y2){
    return sqrt((x2-x1)*(x2-x1) + (y2-y1)*(y2-y1));
}

int main(){
    int N;
    scanf("%d", &N);

    int points[N][2];

    for (int i=0; i<N; i++){
        scanf("%d %d", &points[i][0], &points[i][1]);
    }

    double minDist = FLT_MAX;

    for (int i=0; i<N; i++){
        for (int j =i+1; j<N; j++){
            double dist  = distance(points[i][0], points[i][1], points[j][0], points[j][1]);

            if(dist<minDist){
                minDist = dist;
            }
        }
    }

    printf("%.14f\n", minDist);

    return 0;
}